### PR TITLE
Report results for Broken and Unresolved builds

### PR DESCRIPTION
**Pull request by danidoni**
- [ ] Implement frontend part
- [ ] Implement backend part

### CODE REVIEW

**Comment by danidoni** [comment]
[Report back to SCM build results 'broken' and 'unresolvable'](https://trello.com/c/g5Ud3aZw/1693-report-back-to-scm-build-results-broken-and-unresolvable)

**Comment by dmarcoux** [comment]
[Report back to SCM when build results aren't `failed` and `succeeded`](https://trello.com/c/0N3jHq5M/2257-report-back-to-scm-when-build-results-arent-failed-and-succeeded)

File: src/api/app/jobs/report_to_scm_job.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # We don't properly capitalize SCM in the class name since CreateJob is doing `CLASS_NAME.to_s.camelize.safe_constantize`
 class ReportToScmJob < CreateJob
-  ALLOWED_EVENTS = ['Event::BuildFail', 'Event::BuildSuccess'].freeze
+  ALLOWED_EVENTS = [
+    'Event::BuildBroken',
+    'Event::BuildFail',
+    'Event::BuildSuccess',
```
Comment (coolo) [inline src/api/app/jobs/report_to_scm_job.rb:6]: Note that a build can switch from success to fail and back to unchanged. So if you reported fail, you need to look at BuildUnchanged too

Comment (dmarcoux) [reply src/api/app/jobs/report_to_scm_job.rb:6]: So `Event::BuildUnchanged` is also to be reported as `success`... this is weird, but whatever, this is how it is.

https://github.com/openSUSE/open-build-service/blob/af9297d8d8fb79e14951f657a792474c53179e67/src/api/app/models/event/build_unchanged.rb#L4

Comment (coolo) [reply src/api/app/jobs/report_to_scm_job.rb:6]: Don't shoot the messenger :)

**Comment by wineee** [comment]
It's a useful feature, is it still under development?

**Comment by coolo** [comment]
If I may suggest an approach that might be a little easier to handle. There is a `RepoBuildFinished` event. Once that is in, send broken builds that weren't reported before (i.e. store it somewhere once someone subscribes the event).

Unresolvable state is unfortunately very temporary:
If you have package A, package B depending on A's last built version and then package C requiring B: C is unresolvable after A has built a new version, but once B has built (and refreshed the required version), C is fine again. People asking for unresolvable emails are surely not asking for these temporary states, but about actionable states.

**Comment by wineee** [comment]
> If I may suggest an approach that might be a little easier to handle. There is a `RepoBuildFinished` event. Once that is in, send broken builds that weren't reported before (i.e. store it somewhere once someone subscribes the event).

Does the `RepoBuildFinished` event mean the build broken？ Doesn't build failure/success trigger this event?
（here are no :reason  payload_keys in  `RepoBuildFinished` ）

**Pull request closed without merging**